### PR TITLE
Refactor TalonFXMotor to use functions -> core

### PR DESCRIPTION
**Pull request by drorta**

### CODE REVIEW

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java [hunk 1]
```diff
@@ -48,6 +44,22 @@ public TalonFXMotor(String logPath, Phoenix6DeviceID deviceID, TalonFXConfigurat
 		this(logPath, deviceID, configuration, sysidConfig, null);
 	}
 
+	private TalonFXWrapper createWrapper(Phoenix6DeviceID deviceID) {
+		return Robot.ROBOT_TYPE.isSimulation() ? new TalonFXWrapper(deviceID.ID()) : new TalonFXWrapper(deviceID);
+	}
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java:49]: put in talonfx wrapper

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java [hunk 2]
```diff
@@ -48,6 +44,22 @@ public TalonFXMotor(String logPath, Phoenix6DeviceID deviceID, TalonFXConfigurat
...
+
+	private void applyConfiguration(TalonFXConfiguration configuration) {
+		if (!motor.applyConfiguration(configuration, APPLY_CONFIG_RETRIES).isOK()) {
+			new Alert(Alert.AlertType.ERROR, getLogPath() + "ConfigurationFailed").report();
+		}
+	}
+
+	private Optional<TalonFXSimulation> createTalonFXSimulationOptional(MechanismSimulation simulation, TalonFXConfiguration configuration) {
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXMotor.java:57]:
```suggestion
	private Optional<TalonFXSimulation> createSimulation(MechanismSimulation simulation, TalonFXConfiguration configuration) {
```

**Review by YHerm — CHANGES_REQUESTED** [review]

**Review by hillelv — APPROVED** [review]

**Review by YHerm — APPROVED** [review]
make sure its tested

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java [hunk 3]
```diff
@@ -15,7 +17,7 @@ public TalonFXWrapper(int deviceId) {
 	}
 
 	public TalonFXWrapper(Phoenix6DeviceID ctreDeviceID) {
-		super(ctreDeviceID.ID(), ctreDeviceID.busChain().getChainName());
+		super(ctreDeviceID.ID(), Robot.ROBOT_TYPE.isSimulation() ? BusChain.ROBORIO.getChainName() : ctreDeviceID.busChain().getChainName());
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java:20]: move to function

**Review by YHerm — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java [hunk 4]
```diff
@@ -3,8 +3,11 @@
 import com.ctre.phoenix6.StatusCode;
 import com.ctre.phoenix6.configs.TalonFXConfiguration;
 import com.ctre.phoenix6.hardware.TalonFX;
+import frc.robot.Robot;
+import frc.robot.hardware.phoenix6.BusChain;
 import frc.robot.hardware.phoenix6.Phoenix6DeviceID;
 import frc.robot.hardware.phoenix6.Phoenix6Utils;
+import frc.utils.brakestate.BrakeStateManager;
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java:10]: format my G

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java [hunk 5]
```diff
@@ -15,7 +18,11 @@ public TalonFXWrapper(int deviceId) {
 	}
 
 	public TalonFXWrapper(Phoenix6DeviceID ctreDeviceID) {
-		super(ctreDeviceID.ID(), ctreDeviceID.busChain().getChainName());
+		super(ctreDeviceID.ID(), determineBusChain(ctreDeviceID));
+	}
+
+	private static String determineBusChain(Phoenix6DeviceID ctreDeviceID){
+		return Robot.ROBOT_TYPE.isSimulation() ? BusChain.ROBORIO.getChainName() : ctreDeviceID.busChain().getChainName();
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java:25]: move this to Phoenix6DeviceID

File: src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java [hunk 6]
```diff
@@ -15,7 +15,7 @@ public TalonFXWrapper(int deviceId) {
 	}
 
 	public TalonFXWrapper(Phoenix6DeviceID ctreDeviceID) {
-		super(ctreDeviceID.ID(), ctreDeviceID.busChain().getChainName());
+		super(ctreDeviceID.ID(), Phoenix6DeviceID.determineBusChain(ctreDeviceID));
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/motor/TalonFXWrapper.java:18]: I meant to move also this part to the other class

File: src/main/java/frc/robot/hardware/phoenix6/Phoenix6DeviceID.java [hunk 7]
```diff
@@ -1,9 +1,22 @@
...
+
 public record Phoenix6DeviceID(int ID, BusChain busChain) {
 
 	public Phoenix6DeviceID(int ID) {
 		this(ID, BusChain.ROBORIO);
 	}
 
+	public Phoenix6DeviceID(int ID, BusChain busChain) {
+		this.ID = ID;
+		this.busChain = Objects.equals(determineBusChain(busChain), BusChain.CANIVORE.getChainName()) ? BusChain.CANIVORE : BusChain.ROBORIO;
```
Comment (YoniKiriaty) [inline src/main/java/frc/robot/hardware/phoenix6/Phoenix6DeviceID.java:15]: cant you do 
```suggestion
		this.busChain =determineBusChain(busChain).equals(BusChain.CANIVORE.getChainName()) ? BusChain.CANIVORE : BusChain.ROBORIO;
```

Comment (drorta) [inline src/main/java/frc/robot/hardware/phoenix6/Phoenix6DeviceID.java:15]:
```suggestion
		this.busChain = determineBusChain(busChain);
```

**Review by YoniKiriaty — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/phoenix6/Phoenix6DeviceID.java [hunk 8]
```diff
@@ -1,9 +1,21 @@
 package frc.robot.hardware.phoenix6;
 
+import frc.robot.Robot;
+
+
```
Comment (YHerm) [inline src/main/java/frc/robot/hardware/phoenix6/Phoenix6DeviceID.java:5]: remove space

**Review by hillelv — APPROVED** [review]

**Review by bossyshrimp68 — APPROVED** [review]

**Review by maya1414 — APPROVED** [review]

**Review by YoniKiriaty — APPROVED** [review]

**Pull request merged**